storage/mysql/queue.go

```diff
@@ -103,14 +104,21 @@ func (t *logTreeTX) UpdateSequencedLeaves(ctx context.Context, leaves []*trillia
 			glog.Warningf("Failed to update sequenced leaves: %s", err)
 			return err
 		}
+
+		qe, ok := t.dequeued[string(leaf.LeafIdentityHash)]
+		if !ok {
+			return fmt.Errorf("attempting to assign unknown merkleleafhash %x", leaf.MerkleLeafHash)
+		}
+		dequeuedLeaves = append(dequeuedLeaves, qe)
```

Comment:
Do we expect that after this loop finishes all leaves in `t.dequeued` will have been copied to `dequeuedLeaves`?

If so could enforce this by removing them from the map and checking it's empty after.

Reply:
I don't think so? Leaves only get properly sequenced upon `UpdateLeaves`

Reply:
I'm trying to decide what the invariants should be and if it could go wrong in new ways. If we update a subset of the leaves and commit then presumably the others will just get returned next time so that should be safe.

Reply:
Given we have another version that does the same thing this is probably ok. The batched queue goes through integration as well as unit tests?

Reply:
yes, integration and unit test